### PR TITLE
Front-end detailed card dialog

### PR DESCRIPTION
**Pull request by webdever12**
### What this PR does (required):
- This PR is the front-end of the detailed card
- When a card is clicked on in the board section, this dialog pops up and allows the user to add [Description, Comment, Attachment, Deadline, etc..]
- There are also **Action** buttons that allow the user to [Move, Delete, Copy, Share] the cards
- At the top of the card, the color should match the card's color and can be changed. 
- **In list...** should be integrated with board column's title
- Every section's code is written in separate folders and then integrated into one main file called **DetailedCardDialog.tsx**
- The video below shows the current functionality

### Screenshots / Videos (required):

- A demo shot of the front-end functionality of the detailed card.

https://user-images.githubusercontent.com/77899847/128801259-7c294c4f-d2b9-423a-94ad-c3e124cab3ec.mp4

### Any information needed to test this feature (required):
- Should be run on **port 3000**

### Any issues with the current functionality (optional):
- The save buttons, color chooser at the top, and column title will need to be integrated with back-end and the board
- The Section and Action buttons will also need to be integrated with the back-end

### CODE REVIEW

File: client/src/components/DetailedCardDialog/AddColor/AddColor.tsx [hunk 1]
```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+import ClickAwayListener from '@material-ui/core/ClickAwayListener';
+import Grow from '@material-ui/core/Grow';
+import Paper from '@material-ui/core/Paper';
+import Popper from '@material-ui/core/Popper';
+import MenuList from '@material-ui/core/MenuList';
+import useStyles from './useStyles';
+import { useState } from 'react';
```
Comment (ajaytomgeorge) [inline client/src/components/DetailedCardDialog/AddColor/AddColor.tsx]: you can use import React, { useState } from 'react'

File: client/src/components/DetailedCardDialog/AddColor/AddColor.tsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+import { IconButton, Grid } from '@material-ui/core';
+import FormatColorFillOutlinedIcon from '@material-ui/icons/FormatColorFillOutlined';
+import { FiberManualRecord } from '@material-ui/icons';
+import { Task as TaskInterface } from '../../../interface/Task';
+
+export default function AddColor() {
+  const classes = useStyles();
+  const colors = ['#FFFFFF', '#FF5D48', '#EDAB1D', '#59B0FF', '#D460F7'];
+  const [open, setOpen] = React.useState(false);
```
Comment (ajaytomgeorge) [inline client/src/components/DetailedCardDialog/AddColor/AddColor.tsx]: You are already importing useState  directly:-)

**Review by ajaytomgeorge — COMMENTED** [review]
@hamidrc  Looks good, @moffatethan  Looks good from my side

**Review by moffatethan — COMMENTED** [review]
Please address Ajay's comments otherwise, this looks good to me

File: client/src/components/DetailedCardDialog/AddColor/AddColor.tsx (hunk 1 above)
Comment (webdever12) [reply client/src/components/DetailedCardDialog/AddColor/AddColor.tsx]: Done

**Review by webdever12 — COMMENTED** [review]
Both comments updated :)

**Review by moffatethan — APPROVED** [review]

**Pull request merged**